src/http/ureq.rs

```diff
@@ -0,0 +1,108 @@
+//! The client implementation for the ureq HTTP client, which is blocking.
+
+use super::{headers, BaseClient, Form, Headers, Query};
+use crate::client::{ClientError, ClientResult, Spotify};
+
+use maybe_async::sync_impl;
+use serde_json::Value;
+use ureq::{Request, Response};
+
+impl ClientError {
+    pub fn from_response(r: ureq::Response) -> Self {
+        ClientError::StatusCode(r.status(), r.status_text().to_string())
+    }
+}
+
+impl Spotify {
+    fn request<D>(
+        &self,
+        request: &mut Request,
+        headers: Option<&Headers>,
+        send_request: D,
+    ) -> ClientResult<String>
+    where
+        D: Fn(&mut Request) -> Response,
```

Comment:
It's just a function to continue building the `Request` like the one in `src/http/reqwest.rs`. You can see its usage later on. In this case, calling something like `send_json` with `ureq` converts the Request into a Response, so that's the function signature needed. Using a closure makes it straightforward to choose between GET/POST/PUT/DELETE, and avoids the `Content::Json` and `Content::Form` enum I had previously, which was admittedly a mess.